src/disco/reducers/discoResults.js

```diff
@@ -1,8 +1,13 @@
-import { DISCO_RESULTS } from 'disco/constants';
+import { LOAD_DISCO_RESULTS } from 'disco/constants';
 
 export default function discoResults(state = [], { type, payload }) {
-  if (type === DISCO_RESULTS) {
-    return payload.results;
+  switch (type) {
+    case LOAD_DISCO_RESULTS: {
+      const { entities, result } = payload;
+      // Unwrap the magical discoResults API schema.
```

Comment:
It is magical because you can't figure out how it transforms the results of the API response just by looking at it.